### DISCUSSION TITLE
Adding pre-release builds to Scoop package manager?

### DISCUSSION THREAD

**Discussion by tech189** [sumatrapdfreader/sumatrapdf#2895] [General]
Hi, I was wondering whether it would be okay to create a package for the pre-release versions of Sumatra PDF available [here](https://www.sumatrapdfreader.org/prerelease) and add it to the [Scoop package manager](https://scoop.sh/)?

I understand that you do not want the builds to be redistributed – Scoop will simply point to the direct download URL, so the builds will not be re-hosted. Scoop will just update the package (a JSON file) every day so that users can easily install the latest pre-release with one command, making testing new builds easier for users.

We already have packages for beta versions of Firefox, VLC, GIMP and the stable release of Sumatra PDF is already available in Scoop and Chocolatey so it would be great to add the pre-release version too!

**Comment by b4imetu** [comment]
Many users are still willing to use the git version of `sumatrapdf`, just like many users are used to using the [`mpv-git` version](https://sourceforge.net/projects/mpv-player-windows/files/64bit/).
But if the manual update is a bit of a hassle. So I hope we can use the scoop package management software to manage/update `sumatrapdf` git version.
Relate to https://github.com/ScoopInstaller/Versions/pull/607 & https://github.com/ScoopInstaller/Versions/pull/695

**Comment by b4imetu** [comment]
@kjk Please consider this ^^

**Reply by GitHubRulesOK** [reply]
the version you nominated 3 weeks ago was the start of a round of unstable versions for two different reasons, SumatraPDF has an inbuilt update notification system that resulted in more support issues than normal due to user uptake without due consideration that there is no enhanced rollback system, I have enough problems keeping track of which cutting edge pre-rel multiple updates are more stable than others.

You may not have noticed all versions prior to 3.3.3 offer that as stable. and 3.4.6 is considered the alternative advanced 64bit (not32) update from that, however 32bit 3.4.6 is more suitable for Latex users !

**Reply by b4imetu** [reply]
For the stable version of `sumatrapdf`, it is already provided in another bucket called `Extras`.
See https://github.com/ScoopInstaller/Extras/blob/master/bucket/sumatrapdf.json
The current version is 3.4.6.

In this `Versions` bucket, I want to provide the unstable pre-release version, named `sumatrapdf-pre`, hoping to give users who need to use the latest version an option.

I think there is no need to worry about the stability of the pre-release. The users of the pre-release version must have the habit of frequent updates.

And with `checkver`, `autoupdate` and `github-actions`, the manifest will automatically synchronize with the official website updates

**Reply by GitHubRulesOK** [reply]
I am concerned at seeing `Rename-Item -NewName` the filenames including pre-rel names are set exactly the way they are for a reason, often renaming an installed variant to the basic exe without the installer checks can be a potential source of conflicts.
the installation does several self-checks based on its name.

**Reply by b4imetu** [reply]
I've updated https://github.com/ScoopInstaller/Versions/pull/695.
Hopefully, this distribution of the pre-release would be allowed.

**Comment by b4imetu** [comment]
Please consider this proposal @kjk
